datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ingest/source/UpsertIngestionSourceResolver.java

```diff
@@ -90,20 +98,22 @@ public CompletableFuture<String> get(final DataFetchingEnvironment environment)
     });
   }
 
-  private DataHubIngestionSourceInfo mapIngestionSourceInfo(final UpdateIngestionSourceInput input) {
+  private DataHubIngestionSourceInfo mapIngestionSourceInfo(final UpdateIngestionSourceInput input, final String ingestionSourceUrnString) {
     final DataHubIngestionSourceInfo result = new DataHubIngestionSourceInfo();
     result.setType(input.getType());
     result.setName(input.getName());
-    result.setConfig(mapConfig(input.getConfig()));
+    result.setConfig(mapConfig(input.getConfig(), ingestionSourceUrnString));
     if (input.getSchedule() != null) {
       result.setSchedule(mapSchedule(input.getSchedule()));
     }
     return result;
   }
 
-  private DataHubIngestionSourceConfig mapConfig(final UpdateIngestionSourceConfigInput input) {
+  private DataHubIngestionSourceConfig mapConfig(final UpdateIngestionSourceConfigInput input, final String ingestionSourceUrnString) {
     final DataHubIngestionSourceConfig result = new DataHubIngestionSourceConfig();
-    result.setRecipe(input.getRecipe());
+    String recipe = input.getRecipe();
```

Comment:
Is recipe a required field?

Reply:
If not, there could be null pointer exception on line 135 in future

Reply:
it appears like recipe is indeed required - but adding a check that it's not null won't hurt anybody